champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/brute/passives/Stampede.java

```diff
@@ -140,10 +142,12 @@ public void updateSpeed() {
                         championsManager.getEffects().addEffect(player, player, EffectTypes.SPEED, getName(), data.getSprintStrength(), 2200, true);
                         player.getWorld().playSound(player.getLocation(), Sound.ENTITY_ZOMBIE_AMBIENT, 2.0F, 0.2F * data.getSprintStrength() + 1.2F);
                         UtilMessage.simpleMessage(player, getClassType().getName(), "Stampede Level: <yellow>%d", data.getSprintStrength());
+                        spawnParticles(player.getLocation(), 30);
                     }
                 }
                 if (data.getSprintStrength() > 0) {
                     championsManager.getEffects().addEffect(player, player, EffectTypes.SPEED, getName(), data.getSprintStrength(), 2200, true);
+                    spawnParticles(player.getLocation(), (data.getSprintStrength() + 3) * 2);
```

Comment:
spawnParticles(player.getLocation(), (data.getSprintStrength());

**Too many particles**

Possibly increase the frequency of this event, or create a separate update event that fires every tick to spawn particles. this will make the particles look smooth but should only spawn 1/2/3 per second depending on the speed level

Reply:
Eh, the idea is to show the level based on the particles. This did give me an idea to show particles in frequency based on speed level.

Reply:
The code I gave you would show level based on particles. If you change the frequency to every tick, it means that at speed 1 it would spawn 1 particle per tick, at speed 2 it would spawn 2 per tick, and at level 3 it would spawn 3 per tick. There is no need for a burst of particles when changing speed levels, there is a different pitched zombie sound for that exact reason, the particles should visually show clearly what speed level the player is at from a distance, which is exactly what this code does while not looking too cluttered.